rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -3102,10 +3064,12 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
      * @return
      */
     @CompileStatic
-    def saveExecutionState( schedId, exId, Map props, AsyncStarted execmap, Map retryContext){
-        Execution.withNewTransaction {
+    def saveCompletedExecution_currentTransaction( schedId, exId, Map props, AsyncStarted execmap, Map retryContext){
```

Comment:
The name of this function is a little strange since this function does more than just saving the execution and it actually starts new transactions. What do you think about naming this function:  `completeExecution` or leaving it as `saveExecutionState` and instead renaming the inner function: `private ExecutionCompleteEvent saveExecutionState_currentTransaction(` to something like `private ExecutionCompleteEvent saveCompletedExecution_currentTransaction(` ?